sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java

```diff
@@ -53,6 +55,9 @@ public void blame(BlameInput input, BlameOutput output) {
     File basedir = input.fileSystem().baseDir();
     try (Repository repo = buildRepository(basedir); Git git = Git.wrap(repo)) {
       File gitBaseDir = repo.getWorkTree();
+      if (Files.isRegularFile(gitBaseDir.toPath().resolve(".git/shallow"))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().isFile()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)